crates/bevy_math/src/cubic_splines.rs

```diff
@@ -397,40 +426,55 @@ impl<P: Point> CubicNurbs<P> {
             });
         }
 
-        let mut weights = weights
+        let weights = weights
             .map(Into::into)
             .unwrap_or_else(|| vec![1.0; control_points_len]);
 
-        let knot_vector: Vec<f32> = knot_vector.map(Into::into).unwrap_or_else(|| {
-            Self::open_uniform_knot_vector(control_points_len)
+        let mut knots: Vec<f32> = knots.map(Into::into).unwrap_or_else(|| {
+            Self::open_uniform_knots(control_points_len)
                 .expect("The amount of control points was checked")
         });
 
-        let knot_vector_expected_length = Self::knot_vector_length(control_points_len);
+        let expected_knots_len = Self::knots_len(control_points_len);
 
-        // Check the knot vector length
-        if knot_vector.len() != knot_vector_expected_length {
-            return Err(CubicNurbsError::InvalidKnotVectorLength {
-                expected: knot_vector_expected_length,
-                provided: knot_vector.len(),
+        // Check the number of knots is correct
+        if knots.len() != expected_knots_len {
+            return Err(CubicNurbsError::KnotsMismatch {
+                expected: expected_knots_len,
+                provided: knots.len(),
             });
         }
 
-        // Check the knot vector for being nondescending (previous elements is less than or equal
+        // Ensure the knots are non-descending (previous elements is less than or equal
         // to the next)
-        if knot_vector.windows(2).any(|win| win[0] > win[1]) {
-            return Err(CubicNurbsError::InvalidKnotVectorValues);
+        if knots.windows(2).any(|win| win[0] > win[1]) {
+            return Err(CubicNurbsError::DescendingKnots);
         }
 
-        // Check the weights vector length
+        // Ensure the knots are non-constant
+        if knots.windows(2).all(|win| win[0] == win[1]) {
+            return Err(CubicNurbsError::ConstantKnots);
+        }
+
+        // Check that the number of weights equals the number of control points
         if weights.len() != control_points_len {
-            return Err(CubicNurbsError::WeightsVectorMismatch {
+            return Err(CubicNurbsError::WeightsMismatch {
                 expected: control_points_len,
                 provided: weights.len(),
             });
         }
 
-        weights = Self::normalize_weights(weights);
+        // To align the evaluation behavior of nurbs with the other splines,
+        // make the intervals between knots form an exact cover of [0, N], where N is
+        // the number of segments of the final curve.
+        let curve_length = (control_points.len() - 3) as f32;
+        let min = knots.first().unwrap().clone();
+        let max = knots.last().unwrap().clone();
```

Comment:
```suggestion
        let min = *knots.first().unwrap();
        let max = *knots.last().unwrap();
```
No need for explicit clone I think